internal/harnesses/k3s/k3s.go

```diff
@@ -121,13 +122,25 @@ func New(id string, cli *provider.DockerClient, opts ...Option) (types.Harness,
 		}
 	}
 
+	extraFiles := make([]provider.File, 0)
+	if "" != k3s.opt.KubeletConfig {
+		kubeletConfigBuf := &bytes.Buffer{}
+		kubeletConfigBuf.WriteString(k3s.opt.KubeletConfig)
```

Comment:
if we type `k3s.opt.KubeletConfig` then we can just marshal to json here

Reply:
`Contents` expects an `io.Reader`, but `[]byte` isn't an `io.Reader`. What did you have in mind here?

Reply:
I don't know that this is a good idea. Marshalling to an object then unmarshalling causes a lot more fields to be written out to the final configuration, and that will end up resulting in non-specified fields showing up in the configuration.